sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -1119,4 +1123,16 @@ private String getLinkName(String sessionId) {
             return existing != null ? existing.getLinkName() : null;
         }
     }
+
+    //TODO: discuss whether we should disallow session id call on peek, receiveDeferred.
+//    private Mono<Boolean> validateSession(String sessionId) {
+//        String optionSessionId = receiverOptions.getSessionId();
+//        if (optionSessionId != null && !optionSessionId.equals(sessionId)) {
+//            return monoError(logger, new IllegalArgumentException(String.format(
+//                "This receiver client is tied to session %s. It can't be used for another %s.",
+//                optionSessionId, sessionId)));
+//        } else {
+//            return Mono.just(true);
+//        }
+//    }
```

Comment:
I guess this is now resolved. So, you might want to update this.

Reply:
Removed this part according to latest requirement.